### PR TITLE
Sygma Client Types

### PR DESCRIPTION
**Pull request by kalambet**
This is the first PR to support Sygma in the Gaming SDK.

This PR **only contains Types and no logic** at all.
Some of the classes might be transformed into structures as logic gets implemented.

With the next PRs I'll be adding each of the Sygma Client functions implementations so it is not too overwhelming.

### CODE REVIEW

**Review by rob1997 — APPROVED** [review]
LGTM!

**Review by sneakzttv — APPROVED** [review]

File: src/ChainSafe.Gaming.SygmaClient/SygmaClient.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            this.contractBuilder = contractBuilder;
+            this.signer = signer;
+            this.sourceChainConfig = sourceChainConfig;
+            this.destinationChainConfig = destinationChainConfig;
+            this.analyticsClient = analyticsClient;
+            this.projectConfig = projectConfig;
+        }
+
+        public bool Initialize()
+        {
```
Comment (kantagara) [inline src/ChainSafe.Gaming.SygmaClient/SygmaClient.cs:36]: Is this necessary? If it's an ilifecycle participant we have WillStartAsync

**Review by kantagara — APPROVED** [review]

File: src/ChainSafe.Gaming.SygmaClient/SygmaClient.cs (hunk 1 above)
Comment (kalambet) [reply src/ChainSafe.Gaming.SygmaClient/SygmaClient.cs:36]: But you are right, method arguments are missing. Let me update it.

File: src/ChainSafe.Gaming.SygmaClient/SygmaClient.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            throw new System.NotImplementedException();
+        }
+
+        public Task<TransferStatus> TransferStatusData(Environment environment, string transactionHash)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public ValueTask WillStartAsync()
```
Comment (oleksandrchainsafe) [inline src/ChainSafe.Gaming.SygmaClient/SygmaClient.cs:69]: Usually, we keep those at the top of the methods list as part of the initialization/termination method group.

File: src/ChainSafe.Gaming.SygmaClient/Types/ConfigUrl.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+namespace ChainSafe.Gaming.SygmaClient.Types
+{
+    public static class ConfigUrl
+    {
+        public static readonly string Devnet = "https://chainbridge-assets-stage.s3.us-east-2.amazonaws.com/shared-config-dev.json";
```
Comment (oleksandrchainsafe) [inline src/ChainSafe.Gaming.SygmaClient/Types/ConfigUrl.cs:5]: You can use `const` instead of `static readonly` for strings.

**Review by oleksandrchainsafe — APPROVED** [review]
👍

File: src/ChainSafe.Gaming.SygmaClient/SygmaClient.cs (hunk 2 above)
Comment (kalambet) [reply src/ChainSafe.Gaming.SygmaClient/SygmaClient.cs:69]: Will change it with the next PR! Thanks!

File: src/ChainSafe.Gaming.SygmaClient/Types/ConfigUrl.cs (hunk 3 above)
Comment (kalambet) [reply src/ChainSafe.Gaming.SygmaClient/Types/ConfigUrl.cs:5]: Will do, thank you!

**Pull request merged**